src/test/java/org/springframework/data/couchbase/repository/N1qlCouchbaseRepositoryIntegrationTests.java

```diff
@@ -90,6 +97,41 @@ public void cleanUp() {
     try { partyRepository.deleteById(KEY_PARTY); } catch (DataRetrievalFailureException e) {}
   }
 
+  @Test
+  public void shouldBeThreadsafe() {
+    // This doesn't guarantee it, but we should catch most thread issues without
+    // taking too long here...
+    int runs = 50;
+    for (int i=0; i<runs; i++) {
+      doShouldBeThreadsafe();
+    }
+  }
+  public void doShouldBeThreadsafe() {
+    int threads = 50;
+    ExecutorService service = Executors.newFixedThreadPool(threads);
+    List<Callable<Boolean>> callables = new ArrayList<>();
+    for (int thread = 0; thread < threads; ++thread) {
+      final int counter = thread;
+      Callable<Boolean> booleanSupplier = () -> {
+        String expectedName = "party like it's 199" + counter%12;
+        String foundName = partyRepository.findByName(expectedName).get(0).getName();
+        System.out.println("expected = " + expectedName  + " found = " + foundName);
```

Comment:
make a real assertEquals and not println so it fails the test if needed and not spam the logs

Reply:
I can just remove this line -- I'd popped it in to make understanding what happened easier, during development.